### PR TITLE
statistics: introduce a new kind of syntax to drop global-stats

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: close #xxx 

Problem Summary: statistics: introduce a new kind of syntax to drop global-stats

### What is changed and how it works?

Before this PR, we already have the syntax `drop stats {table-name} partition {partition-name}` to drop partition-stats.
And if we want to drop global-stats, we now use the keyword `global` to replace `{partition-name}` like `drop stats t partition global`.
But this solution may cause a conflict if a user names a partition as `global`.
To solve this conflict, this PR introduces a new kind of syntax `drop stats {table-name} global`.

Have to merge https://github.com/pingcap/parser/pull/1181 first.

### Check List 

Tests 

- Unit test

### Release note 

- statistics: introduce a new kind of syntax to drop global-stats

### CODE REVIEW

File: executor/simple_test.go [hunk 1]
```diff
@@ -630,7 +630,15 @@ partition by range (a) (
 	tk.MustExec("drop stats t partition p0")
 	checkPartitionStats("global", "p1")
 
-	tk.MustExec("drop stats t partition global")
+	err := tk.ExecToErr("drop stats t partition abcde")
```
Comment (Reminiscent) [inline executor/simple_test.go:633]: Please add a partition which names `global` to table `t` for more tests.

**Review by rebelice — APPROVED** [review]

File: executor/simple_test.go (hunk 1 above)
Comment (qw4990) [reply executor/simple_test.go:634]: A nice comment. It is fixed.

**Review by Reminiscent — APPROVED** [review]
/LGTM

**Pull request merged**